backend.c

```diff
@@ -2432,7 +2432,10 @@ static void run_threads(struct sk_out *sk_out)
 							strerror(ret));
 			} else {
 				pid_t pid;
+				struct fio_file **files;
 				dprint(FD_PROCESS, "will fork\n");
+				files = td->files;
+				read_barrier();
```

Comment:
@axboe, what do you think of this line? Is it reasonable for me to insert `read_barrier` quite after reading `td->files` and before `fork`? I thought that is would not be odd and does not hurt anything but ensure here reading files before forking. But I assume that forking implies full memory barrier, doesn't it?

Reply:
If a read_barrier() is needed for that, something is horribly broken.

Reply:
@axboe, read_barrier() is needed in my opinion for a generic way to say to a compiler that 'td->files' must be fetched before getting forked to prevent the compiler from possible reordering of instructions. It is required because after forking we unlikely by may get td->files to be NULL generally speaking, because upon forking the parent process may unlikely but sleep until the child is done. That is why I have introduced 'read_barrier' just to avoid such a rarely faced but possible situation